planner/core/physical_plans.go

```diff
@@ -864,8 +864,15 @@ type basePhysicalJoin struct {
 	InnerJoinKeys []*expression.Column
 	LeftJoinKeys  []*expression.Column
 	RightJoinKeys []*expression.Column
+	// IsNullEQ is used for cases like Except statement where null key should be matched with null key.
+	// <1,null> is exactly matched with <1,null>, where the null value should not be filtered and
+	// the null is exactly matched with null only. (while in NAAJ null value should also be matched
+	// with other non-null item as well)
```

Comment:
I am confused, null value matched with non-null values?

Reply:
yes, null in (y set), for lhs, it doesn't care what the y actually is.

when match <1, null> probe row to hash table, what you ganna fetch from hash-table? the pattern should be <1, whatever>